### PR TITLE
Better logs for evaluation.

### PR DESCRIPTION
**Pull request by czaloom**

### CODE REVIEW

File: api/valor_api/backend/metrics/detection.py [hunk 1]
```diff
@@ -868,11 +868,19 @@ def compute_detection_metrics(*, db: Session, evaluation_id: int):
         .all()
     )
     model = (
-        db.query(Query(models.Model).filter(prediction_filter).any())  # type: ignore - SQLAlchemy type issue
```
Comment (ntlind) [inline api/valor_api/backend/metrics/detection.py:871]: is `prediction_filter` guaranteed to only include this one check (`models.Model.name == evaluation.model_name`)?

File: api/valor_api/backend/metrics/detection.py [hunk 2]
```diff
@@ -868,11 +868,19 @@ def compute_detection_metrics(*, db: Session, evaluation_id: int):
...
-        .distinct()
+        db.query(models.Model)
+        .where(models.Model.name == evaluation.model_name)
         .one_or_none()
     )
 
+    # verify
+    if not datasets:
+        raise RuntimeError("No datasets returned by datum filter.")
```
Comment (ntlind) [inline api/valor_api/backend/metrics/detection.py:878]: could you add a test to check that this branching logic works as expected?

File: api/valor_api/backend/metrics/metric_utils.py [hunk 3]
```diff
@@ -294,8 +294,10 @@ def wrapper(*args, **kwargs):
             core.set_evaluation_status(
                 db, evaluation_id, enums.EvaluationStatus.FAILED
             )
-            logger.info(
-                f"Evaluation `{evaluation_id}` failed with error: {str(e)}"
+            logger.error(
+                "Valor Evaluation Exception",
```
Comment (ntlind) [inline api/valor_api/backend/metrics/metric_utils.py:298]: should we include the `evaluation_id` in the error message?

File: api/valor_api/backend/metrics/detection.py (hunk 1 above)
Comment (czaloom) [reply api/valor_api/backend/metrics/detection.py:871]: Reverting this

**Review by ntlind — APPROVED** [review]

**Pull request merged**